FWCore/Concurrency/test/test_threadSafeOutputFileStream.cpp

```diff
@@ -0,0 +1,31 @@
+#include "FWCore/Concurrency/interface/ThreadSafeOutputFileStream.h"
+
+#include <sstream>
+#include <thread>
+#include <vector>
+
+namespace {
+  void
+  logToFile(unsigned const thread_index, edm::ThreadSafeOutputFileStream& f)
+  {
```

Comment:
I suggest adding a synchronization point (e.g. decreasing a global `std::atomic<int>`) so that all threads actually start work at the same time. Else, it is very easy for one thread to do all its work before the next thread actually starts.
```
namespace { std::atomic<int> nWaitingForThreads; }
```
In main
```
nWaitingForThreads = threads.size();
```
in logToFile
```
--nWaitingForThreads;
while( 0 != nWaitingForThreads);
```